# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTexture.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLTexture.cpp
--- a/dom/canvas/WebGLTexture.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLTexture.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -69,11 +69,6 @@
 
 ////////////////////////////////////////
 
-JSObject* WebGLTexture::WrapObject(JSContext* cx,
-                                   JS::Handle<JSObject*> givenProto) {
-  return dom::WebGLTexture_Binding::Wrap(cx, this, givenProto);
-}
-
 WebGLTexture::WebGLTexture(WebGLContext* webgl, GLuint tex)
     : WebGLRefCountedObject(webgl),
       mGLName(tex),
@@ -824,22 +819,23 @@
   PopulateMipChain(maxLevel);
 }
 
-JS::Value WebGLTexture::GetTexParameter(TexTarget texTarget, GLenum pname) {
+MaybeWebGLVariant WebGLTexture::GetTexParameter(TexTarget texTarget,
+                                                GLenum pname) {
   GLint i = 0;
   GLfloat f = 0.0f;
 
   switch (pname) {
     case LOCAL_GL_TEXTURE_BASE_LEVEL:
-      return JS::NumberValue(mBaseMipmapLevel);
+      return AsSomeVariant(mBaseMipmapLevel);
 
     case LOCAL_GL_TEXTURE_MAX_LEVEL:
-      return JS::NumberValue(mMaxMipmapLevel);
+      return AsSomeVariant(mMaxMipmapLevel);
 
     case LOCAL_GL_TEXTURE_IMMUTABLE_FORMAT:
-      return JS::BooleanValue(mImmutable);
+      return AsSomeVariant(mImmutable);
 
     case LOCAL_GL_TEXTURE_IMMUTABLE_LEVELS:
-      return JS::NumberValue(uint32_t(mImmutableLevelCount));
+      return AsSomeVariant(uint32_t(mImmutableLevelCount));
 
     case LOCAL_GL_TEXTURE_MIN_FILTER:
     case LOCAL_GL_TEXTURE_MAG_FILTER:
@@ -849,13 +845,13 @@
     case LOCAL_GL_TEXTURE_COMPARE_MODE:
     case LOCAL_GL_TEXTURE_COMPARE_FUNC:
       mContext->gl->fGetTexParameteriv(texTarget.get(), pname, &i);
-      return JS::NumberValue(uint32_t(i));
+      return AsSomeVariant(uint32_t(i));
 
     case LOCAL_GL_TEXTURE_MAX_ANISOTROPY_EXT:
     case LOCAL_GL_TEXTURE_MAX_LOD:
     case LOCAL_GL_TEXTURE_MIN_LOD:
       mContext->gl->fGetTexParameterfv(texTarget.get(), pname, &f);
-      return JS::NumberValue(float(f));
+      return AsSomeVariant(float(f));
 
     default:
       MOZ_CRASH("GFX: Unhandled pname.");
@@ -1077,11 +1073,4 @@
   InvalidateCaches();
 }
 
-////////////////////////////////////////////////////////////////////////////////
-
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE_0(WebGLTexture)
-
-NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(WebGLTexture, AddRef)
-NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(WebGLTexture, Release)
-
 }  // namespace mozilla